engine/routes.py

```diff
@@ -49,7 +49,10 @@ def particles():
     validate(input_data)
     start = input_data.get('start', 1)
     stop = input_data.get('stop', ntplib.system_to_ntp_time(time.time()))
-    return Response(util.calc.get_particles(input_data.get('streams'), start, stop, input_data.get('coefficients', {})),
+    limit = input_data.get('limit', 0)
+    if limit <= 0 :
```

Comment:
This code will break if we pass in a string here; we should either try to catch all error conditions or none. Maybe isinstance(limit, int), if not set limit to None?